web/core/components/workspace/sidebar/favorites/favorite-items/common/favorite-item-quick-action.tsx

```diff
@@ -0,0 +1,48 @@
+"use client";
+import React, { FC } from "react";
+import { MoreHorizontal, Star } from "lucide-react";
+import { IFavorite } from "@plane/types";
+// ui
+import { CustomMenu } from "@plane/ui";
+// helpers
+import { cn } from "@/helpers/common.helper";
+
+type Props = {
+  ref: React.MutableRefObject<HTMLDivElement | null>;
+  isMenuActive: boolean;
+  favorite: IFavorite;
+  setIsMenuActive: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
Accept an `onChange` prop instead of accepting `setState`. This will give more flexibility to add additional logic, if needed, in future.